packages/cli/src/commands/start.ts

```diff
@@ -78,91 +83,127 @@ async function askQuestions({ template: defaultTemplate }: Pick<StartOptions, "t
-        choices,
-      },
-    ]);
+    const templateType = await p.select({
+      message: "Which project type do you want to start?",
+      options: [
+        { value: "standalone", label: `Standalone ${blueBright("Agent Server")}` },
+        { value: "nodejs", label: `NodeJS ${blueBright("Agent Server")}` },
+        { value: "nestjs", label: `NestJS ${blueBright("Agent Server")}` },
+        { value: "react", label: `React ${blueBright("Application")}` },
+        { value: "nestjs+react", label: `NestJS + React ${blueBright("Agent Server + Client Application")}` },
+      ] as const satisfies { value: StarterTemplate; label: string }[],
+    });
+
+    if (p.isCancel(templateType)) {
+      process.exit(0);
+    }
+
     context.template = templateType;
   }
 
   // Ask for port
   if (context.template !== "standalone") {
-    const { port } = await inquirer.prompt<{ port: number }>([
-      {
-        type: "input",
-        name: "port",
-        message: "Server Port(if project is client app, this port mean ws server port):",
-        default: 3000,
+    const port = await p.text({
+      message: "Server Port(if project is client app, this port mean ws server port):",
+      initialValue: "3000",
+      validate(value) {
+        if (Number.isNaN(Number.parseInt(value))) {
+          return "Port must be an integer";
+        }
+        return undefined;
       },
-    ]);
-    context.port = port;
+    });
+    if (p.isCancel(port)) {
+      process.exit(0);
+    }
+    context.port = Number(port);
   }
 
   // ask if you need connectors
   // if template is react, we don't need connectors
   if (context.template !== "react") {
     const connectors = await getConnectors();
     const sortedConnectors = connectors.sort((a, b) => a.displayName.localeCompare(b.displayName));
-    const serviceChoices = sortedConnectors.map(({ displayName, serviceName }) => ({ name: displayName, value: serviceName }));
-    const { services } = await inquirer.prompt<{ services: Service[] }>([
-      {
-        type: "checkbox",
-        name: "services",
-        message: "Which connectors do you want to include?",
-        choices: serviceChoices,
-      },
-    ]);
+    const serviceChoices = sortedConnectors.map(({ displayName, serviceName }) => ({ label: displayName, value: serviceName }));
+    const services = await p.multiselect({
+      message: "Which connectors do you want to include?",
+      options: serviceChoices,
+      required: false,
+    });
+    if (p.isCancel(services)) {
+      process.exit(0);
+    }
     context.services = services;
   }
 
   // Ask for openAI key
   {
-    const { openAIKey } = await inquirer.prompt<{ openAIKey: string | null }>([
-      {
-        type: "input",
-        name: "openAIKey",
-        message: "Please enter your OPENAI API key (optional):",
-      },
-    ]);
-    context.openAIKey = openAIKey;
+    const isConfirm = await p.confirm({
```

Comment:
Nit.

I think optional is better. Of course, this is just my personal opinion, and I'm not sure what others think. I don't use the CLI much, so I'm not sure how it's usually done.

Generally, other CLIs also use this kind of logic, so it should be fine.

Reply:
![SCR-20250325-kkme](https://github.com/user-attachments/assets/951894d6-c62a-4112-becb-df7bdf06025f)
When I use another tool(mastra), I found this prompt and I like it because I thought this was more explicit than leaving it blank.
It's a tricky one, I think it could go either way.

Reply:
well... In that case, let's proceed with the PR as it is without any changes!
thank you for your suggestion

Reply:
Thank you! we can change it after we get a feedback! Thanks!
I understand your point of view - perhaps we should have separated the PRs. Thank you very much.

Reply:
@sunrabbit123 I modified with `promisify` and update mock function! please review it!

Reply:
we reduced 10kb to remove tinyexec!

Reply:
Anyway, thank you!

Maybe I misunderstood because I mixed up several translators!

Reply:
oooh.... this function return type can't be null, only T or null
so i wrote strict compare
Rather i write loose compare when related to open ai api

Reply:
not here? I think. But it is understandable!

Reply:
omg... Thank you for your understanding.